### PR TITLE
CRM-19871 Default membership type not selected

### PR DESCRIPTION
**Pull request by konadave**
* [CRM-19871: Default Membership Type not selected - Contribution Form](https://issues.civicrm.org/jira/browse/CRM-19871)

### CODE REVIEW

File: CRM/Price/BAO/PriceField.php [hunk 1]
```diff
@@ -136,7 +136,7 @@ public static function create(&$params) {
           'membership_type_id' => CRM_Utils_Array::value($index, CRM_Utils_Array::value('membership_type_id', $params), NULL),
           'weight' => $params['option_weight'][$index],
           'is_active' => 1,
-          'is_default' => CRM_Utils_Array::value($params['option_weight'][$index], $defaultArray) ? $defaultArray[$params['option_weight'][$index]] : 0,
+          'is_default' => CRM_Utils_Array::value($params['membership_type_id'][$index], $defaultArray) ? $defaultArray[$params['membership_type_id'][$index]] : 0,
```
Comment (pradpnayak) [inline CRM/Price/BAO/PriceField.php]: This is w.r.t Membership Type, but Price fields are also used for Events and Contribution. I don't find any issues w.r.t to defaulting of membership type

Comment (konadave) [inline CRM/Price/BAO/PriceField.php]: It's not an issue so long as membership_type_id == weight. When you start changing the weight of the membership types is when it becomes an issue. We've seen this on two sites now, and this fixed them both.

Comment (pradpnayak) [inline CRM/Price/BAO/PriceField.php]: Agreed, but the above change will work only if priceset is related to Membership

**Comment by eileenmcnaughton** [comment]
I'm going to close this as it appears to be an incomplete fix (fixes some scenarios but from @pradpnayak comment breaks another.
Since the PR is nearly a year old it's better to close & re-open if addressed

**Pull request closed without merging**